### ISSUE TITLE
bug: passing 'undefined as never' into function with 'LazyArg' compiles wrong code

### ISSUE THREAD

**Issue by wesselvdv** [ts-plus/typescript#186]
```typescript
function f<X>(x: LazyArg<X>): X {
   return x()
}

f(undefined as never)
f(undefined as void)
```

compiles into:

```javascript
function f(x) {
    return x();
}
f(undefined);
f(() => undefined);
```

This is due to `never extends () => any`, but we should show a diagnostic that this is not correct.
